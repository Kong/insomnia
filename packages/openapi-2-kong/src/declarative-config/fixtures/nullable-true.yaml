openapi: '3.0.2'
x-kong-name: cktest_cards_api
info:
  title: FTI-3278 Test
  version: '1.0'
  description: A sample API specification for FTI-3278
  contact:
    email: support@insomnia.rest

servers:
- url: https://alb-dev01.openapi-test.merlin/

tags:
  - name: oauth2-credential-flow

paths:
  /cktest/cktest_cards/api1:
    get:
      description: test oauth2
      operationId: get_api1
      tags:
        - oauth2-credential-flow

      responses:
        '200':
          description: OK

    post:
      description: test oauth2
      operationId: post_api1
      x-kong-plugin-request-validator:
        enabled: true
        config:
          verbose_response: true
      requestBody:
        content:
          application/json:
            schema:
              "$ref": "#/components/schemas/request-body-nullable"
      responses:
        '200':
          description: OK

      tags:
        - oauth2-credential-flow


components:
  schemas:
    request-body-nullable:
      type: object
      title: request-body-nullable
      description: This object is sent in post application request.
      required:
        - redirectUri
      properties:
        redirectUri:
          type: string
          nullable: true
