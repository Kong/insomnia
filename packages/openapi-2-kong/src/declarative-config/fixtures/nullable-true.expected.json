{
    "_format_version": "1.1",
    "services": [
        {
            "name": "cktest_cards_api",
            "protocol": "https",
            "host": "alb-dev01.openapi-test.merlin",
            "port": 443,
            "path": "/",
            "plugins": [],
            "routes": [
                {
                    "tags": [
                        "OAS3_import",
                        "OAS3file_nullable-true.yaml"
                    ],
                    "name": "cktest_cards_api-get_api1",
                    "methods": [
                        "GET"
                    ],
                    "paths": [
                        "/cktest/cktest_cards/api1$"
                    ],
                    "strip_path": false
                },
                {
                    "tags": [
                        "OAS3_import",
                        "OAS3file_nullable-true.yaml"
                    ],
                    "name": "cktest_cards_api-post_api1",
                    "methods": [
                        "POST"
                    ],
                    "paths": [
                        "/cktest/cktest_cards/api1$"
                    ],
                    "strip_path": false,
                    "plugins": [
                        {
                            "name": "request-validator",
                            "config": {
                                "version": "draft4",
                                "body_schema": "{\"type\":\"object\",\"title\":\"request-body-nullable\",\"description\":\"This object is sent in post application request.\",\"required\":[\"redirectUri\"],\"properties\":{\"redirectUri\":{\"type\":[\"string\",\"null\"],\"nullable\":true}}}",
                                "allowed_content_types": [
                                    "application/json"
                                ],
                                "verbose_response": true
                            },
                            "tags": [
                                "OAS3_import",
                                "OAS3file_nullable-true.yaml"
                            ],
                            "enabled": true
                        }
                    ]
                }
            ],
            "tags": [
                "OAS3_import",
                "OAS3file_nullable-true.yaml"
            ]
        }
    ]
}
